src/libs/geolocation.js

```diff
@@ -10,20 +11,24 @@ const geolocationPermissions = {
 let hasPermissionModalOpenedOnce = false;
 
 export async function showGeolocationModalIfNeeded() {
+  if (!isMobileDevice()) {
+    return false;
+  }
+
+  // Some browsers (Safari, etc) do not implement Permissions API
   if (!window.navigator.permissions) {
-    // Some browsers (Safari, etc) do not implement Permissions API
-    return;
+    return false;
   }
 
+  // granted or denied
   const p = await window.navigator.permissions.query({ name: 'geolocation' });
   if (p.state !== geolocationPermissions.PROMPT) {
-    // allowed or denied
-    return;
+    return p.state === geolocationPermissions.GRANTED;
```

Comment:
Not sure, but if permission has already been granted, we shouldn't show the modal again, right? So we should return false here, no?

Reply:
No, for me it's right the way it is:

If geolocationPermissions.GRANTED is false, the user refused to be located, showGeolocationModalIfNeeded returns true, and the function that calls it (geolocate) does no native request.

If geolocationPermissions.GRANTED is true, the user accepted to be located, showGeolocationModalIfNeeded returns true, and geolocate continues by doing the native request.

Otherwise, it means the permission is pending, so the modal is shown by showGeolocationModalIfNeeded